### PR TITLE
HBASE-28058 Adjust the order of acquiring the takeSnapshot locks to r…esolve the deadlock issue.

### PR DESCRIPTION
**Pull request by hiping-tech**

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 25s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ branch-3 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 58s |  branch-3 passed  |
| +1 :green_heart: |  compile  |   2m 28s |  branch-3 passed  |
| +1 :green_heart: |  checkstyle  |   0m 35s |  branch-3 passed  |
| +1 :green_heart: |  spotless  |   0m 42s |  branch has no errors when running spotless:check.  |
| +1 :green_heart: |  spotbugs  |   1m 28s |  branch-3 passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 34s |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 25s |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 25s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 30s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  hadoopcheck  |   9m 32s |  Patch does not cause any errors with Hadoop 3.2.4 3.3.5.  |
| -1 :x: |  spotless  |   0m 34s |  patch has 22 errors when running spotless:check, run spotless:apply to fix.  |
| +1 :green_heart: |  spotbugs  |   1m 32s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m  9s |  The patch does not generate ASF License warnings.  |
|  |   |  31m 43s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5381/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/5381 |
| Optional Tests | dupname asflicense javac spotbugs hadoopcheck hbaseanti spotless checkstyle compile |
| uname | Linux 8d29fb33d6a8 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | branch-3 / de1a5647bc |
| Default Java | Eclipse Adoptium-11.0.17+8 |
| spotless | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5381/1/artifact/yetus-general-check/output/patch-spotless.txt |
| Max. process+thread count | 79 (vs. ulimit of 30000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5381/1/console |
| versions | git=2.34.1 maven=3.8.6 spotbugs=4.7.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 25s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  2s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ branch-3 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 58s |  branch-3 passed  |
| +1 :green_heart: |  compile  |   0m 44s |  branch-3 passed  |
| +1 :green_heart: |  shadedjars  |   5m 10s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 24s |  branch-3 passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 44s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 45s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 45s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   5m  8s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 23s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 211m  1s |  hbase-server in the patch passed.  |
|  |   | 233m 29s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5381/1/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/5381 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 64c7ebf803fe 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | branch-3 / de1a5647bc |
| Default Java | Eclipse Adoptium-11.0.17+8 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5381/1/testReport/ |
| Max. process+thread count | 4653 (vs. ulimit of 30000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5381/1/console |
| versions | git=2.34.1 maven=3.8.6 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 25s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ branch-3 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 33s |  branch-3 passed  |
| +1 :green_heart: |  compile  |   0m 35s |  branch-3 passed  |
| +1 :green_heart: |  shadedjars  |   4m 54s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 22s |  branch-3 passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 13s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 35s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 35s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   4m 51s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 21s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 216m 26s |  hbase-server in the patch passed.  |
|  |   | 237m  8s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5381/1/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/5381 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 108194a6608d 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | branch-3 / de1a5647bc |
| Default Java | Temurin-1.8.0_352-b08 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5381/1/testReport/ |
| Max. process+thread count | 4597 (vs. ulimit of 30000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5381/1/console |
| versions | git=2.34.1 maven=3.8.6 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by frostruan** [comment]
this is indeed a problem. nice finding ! Thanks for the PR.

+1

Also, do you think if we can just make `submitSnapshotProcedure` a `synchronized` method?

File: hbase-server/src/main/java/org/apache/hadoop/hbase/master/snapshot/SnapshotManager.java [hunk 1]
```diff
@@ -670,11 +670,13 @@ public void takeSnapshot(SnapshotDescription snapshot) throws IOException {
     }
   }
 
-  public synchronized long takeSnapshot(SnapshotDescription snapshot, long nonceGroup, long nonce)
+  public long takeSnapshot(SnapshotDescription snapshot, long nonceGroup, long nonce)
     throws IOException {
     this.takingSnapshotLock.readLock().lock();
     try {
-      return submitSnapshotProcedure(snapshot, nonceGroup, nonce);
+      synchronized(this){
```
Comment (Apache9) [inline hbase-server/src/main/java/org/apache/hadoop/hbase/master/snapshot/SnapshotManager.java:677]: I think we can just add a synchronized modifier on the submitSnapshotProcedure method? To align with other methods in the class?

**Comment by frostruan** [comment]
hi @hiping-tech  any update here ?

File: hbase-server/src/main/java/org/apache/hadoop/hbase/master/snapshot/SnapshotManager.java (hunk 1 above)
Comment (hiping-tech) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/master/snapshot/SnapshotManager.java]: Thank you for your review and valuable suggestions. I have made the necessary changes based on the suggestions provided.

**Comment by hiping-tech** [comment]
hi @frostruan thanks for you excellent suggestions.  I have made the necessary changes.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 29s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ branch-3 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 51s |  branch-3 passed  |
| +1 :green_heart: |  compile  |   2m 22s |  branch-3 passed  |
| +1 :green_heart: |  checkstyle  |   0m 35s |  branch-3 passed  |
| +1 :green_heart: |  spotless  |   0m 40s |  branch has no errors when running spotless:check.  |
| +1 :green_heart: |  spotbugs  |   1m 28s |  branch-3 passed  |
| -0 :warning: |  patch  |   1m 34s |  Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 29s |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 51s |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 51s |  the patch passed  |
| -0 :warning: |  checkstyle  |   0m 32s |  hbase-server: The patch generated 1 new + 4 unchanged - 0 fixed = 5 total (was 4)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  hadoopcheck  |   9m 43s |  Patch does not cause any errors with Hadoop 3.2.4 3.3.5.  |
| -1 :x: |  spotless  |   0m 33s |  patch has 24 errors when running spotless:check, run spotless:apply to fix.  |
| +1 :green_heart: |  spotbugs  |   1m 27s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 10s |  The patch does not generate ASF License warnings.  |
|  |   |  31m 46s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5381/2/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/5381 |
| Optional Tests | dupname asflicense javac spotbugs hadoopcheck hbaseanti spotless checkstyle compile |
| uname | Linux c2dac48ce6aa 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | branch-3 / ef3eea8a5f |
| Default Java | Eclipse Adoptium-11.0.17+8 |
| checkstyle | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5381/2/artifact/yetus-general-check/output/diff-checkstyle-hbase-server.txt |
| spotless | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5381/2/artifact/yetus-general-check/output/patch-spotless.txt |
| Max. process+thread count | 77 (vs. ulimit of 30000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5381/2/console |
| versions | git=2.34.1 maven=3.8.6 spotbugs=4.7.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 28s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  2s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ branch-3 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m  1s |  branch-3 passed  |
| +1 :green_heart: |  compile  |   0m 46s |  branch-3 passed  |
| +1 :green_heart: |  shadedjars  |   5m 13s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 25s |  branch-3 passed  |
| -0 :warning: |  patch  |   5m 45s |  Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 44s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 45s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 45s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   5m  7s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 22s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 210m 48s |  hbase-server in the patch passed.  |
|  |   | 233m 28s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5381/2/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/5381 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 9466d5372d6a 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | branch-3 / ef3eea8a5f |
| Default Java | Eclipse Adoptium-11.0.17+8 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5381/2/testReport/ |
| Max. process+thread count | 4670 (vs. ulimit of 30000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5381/2/console |
| versions | git=2.34.1 maven=3.8.6 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 27s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ branch-3 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 24s |  branch-3 passed  |
| +1 :green_heart: |  compile  |   0m 36s |  branch-3 passed  |
| +1 :green_heart: |  shadedjars  |   4m 55s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 21s |  branch-3 passed  |
| -0 :warning: |  patch  |   5m 24s |  Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 10s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 35s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 35s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   4m 53s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 20s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 219m 20s |  hbase-server in the patch passed.  |
|  |   | 239m 56s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5381/2/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/5381 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux d3defa2f9947 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | branch-3 / ef3eea8a5f |
| Default Java | Temurin-1.8.0_352-b08 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5381/2/testReport/ |
| Max. process+thread count | 4612 (vs. ulimit of 30000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5381/2/console |
| versions | git=2.34.1 maven=3.8.6 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Review by Apache9 — APPROVED** [review]

**Comment by Apache9** [comment]
Please run `mvn spotless:apply` to fix the style error?

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 26s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ branch-3 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 33s |  branch-3 passed  |
| +1 :green_heart: |  compile  |   2m 24s |  branch-3 passed  |
| +1 :green_heart: |  checkstyle  |   0m 33s |  branch-3 passed  |
| +1 :green_heart: |  spotless  |   0m 39s |  branch has no errors when running spotless:check.  |
| +1 :green_heart: |  spotbugs  |   1m 22s |  branch-3 passed  |
| -0 :warning: |  patch  |   1m 28s |  Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 31s |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 25s |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 25s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 32s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  hadoopcheck  |   9m 14s |  Patch does not cause any errors with Hadoop 3.2.4 3.3.5.  |
| -1 :x: |  spotless  |   0m 39s |  patch has 1 errors when running spotless:check, run spotless:apply to fix.  |
| +1 :green_heart: |  spotbugs  |   1m 30s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m  8s |  The patch does not generate ASF License warnings.  |
|  |   |  30m 54s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5381/3/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/5381 |
| Optional Tests | dupname asflicense javac spotbugs hadoopcheck hbaseanti spotless checkstyle compile |
| uname | Linux 3e10d07887c6 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | branch-3 / 05449bd763 |
| Default Java | Eclipse Adoptium-11.0.17+8 |
| spotless | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5381/3/artifact/yetus-general-check/output/patch-spotless.txt |
| Max. process+thread count | 79 (vs. ulimit of 30000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5381/3/console |
| versions | git=2.34.1 maven=3.8.6 spotbugs=4.7.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Review by pankaj72981 — APPROVED** [review]

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 25s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ branch-3 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 31s |  branch-3 passed  |
| +1 :green_heart: |  compile  |   2m 25s |  branch-3 passed  |
| +1 :green_heart: |  checkstyle  |   0m 33s |  branch-3 passed  |
| +1 :green_heart: |  spotless  |   0m 38s |  branch has no errors when running spotless:check.  |
| +1 :green_heart: |  spotbugs  |   1m 27s |  branch-3 passed  |
| -0 :warning: |  patch  |   1m 33s |  Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 29s |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 55s |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 55s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 37s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  hadoopcheck  |   9m 38s |  Patch does not cause any errors with Hadoop 3.2.4 3.3.5.  |
| +1 :green_heart: |  spotless  |   0m 36s |  patch has no errors when running spotless:check.  |
| +1 :green_heart: |  spotbugs  |   1m 26s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m  7s |  The patch does not generate ASF License warnings.  |
|  |   |  31m 41s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5381/4/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/5381 |
| Optional Tests | dupname asflicense javac spotbugs hadoopcheck hbaseanti spotless checkstyle compile |
| uname | Linux a9eacba2ee10 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | branch-3 / 05449bd763 |
| Default Java | Eclipse Adoptium-11.0.17+8 |
| Max. process+thread count | 79 (vs. ulimit of 30000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5381/4/console |
| versions | git=2.34.1 maven=3.8.6 spotbugs=4.7.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 26s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  2s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ branch-3 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 44s |  branch-3 passed  |
| +1 :green_heart: |  compile  |   0m 45s |  branch-3 passed  |
| +1 :green_heart: |  shadedjars  |   5m  8s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 22s |  branch-3 passed  |
| -0 :warning: |  patch  |   5m 37s |  Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 44s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 44s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 44s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   5m  8s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 22s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 213m 10s |  hbase-server in the patch failed.  |
|  |   | 235m 38s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5381/4/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/5381 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 9db42286e30f 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | branch-3 / 05449bd763 |
| Default Java | Eclipse Adoptium-11.0.17+8 |
| unit | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5381/4/artifact/yetus-jdk11-hadoop3-check/output/patch-unit-hbase-server.txt |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5381/4/testReport/ |
| Max. process+thread count | 4681 (vs. ulimit of 30000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5381/4/console |
| versions | git=2.34.1 maven=3.8.6 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 23s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ branch-3 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 13s |  branch-3 passed  |
| +1 :green_heart: |  compile  |   0m 35s |  branch-3 passed  |
| +1 :green_heart: |  shadedjars  |   4m 54s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 21s |  branch-3 passed  |
| -0 :warning: |  patch  |   5m 22s |  Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 16s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 35s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 35s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   4m 53s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 22s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 217m  3s |  hbase-server in the patch passed.  |
|  |   | 237m 54s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5381/4/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/5381 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux e9668eb01f13 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | branch-3 / 05449bd763 |
| Default Java | Temurin-1.8.0_352-b08 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5381/4/testReport/ |
| Max. process+thread count | 4536 (vs. ulimit of 30000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5381/4/console |
| versions | git=2.34.1 maven=3.8.6 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache9** [comment]
Oh, just notice that the PR is opened against branch-3...

Next time please open an PR against master branch. In HBase we will first apply changes to master branch, and then cherry-pick them back to other branches.

Thanks.

**Comment by hiping-tech** [comment]
@Apache9 Thanks for your reminder. I will pay attention next time.

**Pull request merged**